src/charm.py

```diff
@@ -18,79 +18,56 @@
 )
 
 
+class CheckFailed(Exception):
+    """ Raise this exception if one of the checks in main fails. """
+
+    def __init__(self, msg, status=None):
+        super().__init__()
+
+        self.msg = msg
+        self.status = status
+
+
 class Operator(CharmBase):
     _stored = StoredState()
```

Comment:
```suggestion
```

Reply:
@knkski made this comment to me on another charm.  I think this comes from a template, but do we need it?

Reply:
No we don't, I'll get rid of it.